scripts/firedrake-install

```diff
@@ -1030,6 +1031,31 @@ def build_and_install_petsc():
                 check_call(["rm", "-f", deletefile])
 
 
+def build_and_install_petsc4py():
+    with environment(**blas):
+        if os.access(get_petsc4py_dir(), os.W_OK):
+            pipargs = ["-vvv", "--ignore-installed", get_petsc4py_dir()]
+            check_output(pipinstall + pipargs)
+        else:
+            log.info("Could not perform in-place install of petsc4py in current location")
+            log.info("Copying petsc4py to virtualenv and trying again...")
+            petsc4py_copy = os.path.join(os.environ['VIRTUAL_ENV'], 'src', 'petsc4py')
+            if os.path.exists(petsc4py_copy):
+                petsc4py_backup = os.path.join(os.environ['VIRTUAL_ENV'], 'src', 'petsc4py.backup')
+                if os.path.exists(petsc4py_backup):
+                    raise OSError('`{}` exists remove or rename this file/directory and try running this command again'.format(petsc4py_backup))
+                shutil.move(petsc4py_copy, petsc4py_backup)
+            shutil.copytree(get_petsc4py_dir(), petsc4py_copy, dirs_exist_ok=True)
+            S_644 = stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IROTH
+            S_755 = stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR | stat.S_IRGRP | stat.S_IXGRP | stat.S_IROTH | stat.S_IXOTH
```

Comment:
Is there not a nicer way to do this? I came across this [octal representation](https://docs.python.org/3/library/pathlib.html#pathlib.Path.chmod) which might be cleaner.